nengo/builder/signal.py

```diff
@@ -241,8 +241,14 @@ def reshape(self, *shape):
         if len(shape) == 1:
             shape = shape[0]  # in case a tuple is passed in
         initial_value = self.initial_value.view()
-        initial_value.shape = shape  # raises AttributeError if cannot
-                                     # reshape without copying.
+        try:
+            # this raises AttributeError if cannot reshape without copying
+            initial_value.shape = shape
+        except AttributeError:
+            raise SignalError(
+                "Reshaping %s to %s would require the array to be copied "
+                "(because it is not contiguous), which is not supported" % (
+                    self, shape))
```

Comment:
This is slightly misleading, since some non-contiguous arrays can be reshaped, depending on the shape and the way that they are non-contiguous. What if we said: "due to the way it is not contiguous" instead of "because ...". Is that any clearer? Or would most people just read that as a more-poorly-worded version of the current statement?

Reply:
I would read the "because" as applying to "require the array to be copied".  I.e., the reason it is being copied is because it is not contiguous, which is true.  I don't think it implies that all non-contiguous arrays would be copied.  But I also don't think it's that important, so feel free to change it to whatever wording makes sense to you.